mkt/ratings/models.py

```diff
@@ -18,10 +19,30 @@
 
 class ReviewManager(ManagerBase):
 
+    def __init__(self, include_deleted=False):
+        super(ReviewManager, self).__init__()
+        self.include_deleted = include_deleted
+
+    def get_queryset(self):
+        qs = super(ReviewManager, self).get_queryset()
+        qs = qs._clone(klass=ReviewQuerySet)
+        if not self.include_deleted:
+            qs = qs.exclude(deleted=True).exclude(reply_to__deleted=True)
+        return qs
+
     def valid(self):
         """Get all reviews that aren't replies."""
-        # Use extra because Django wants to do a LEFT OUTER JOIN.
-        return self.extra(where=['reply_to IS NULL'])
+        return self.filter(reply_to__isnull=True)
```

Comment:
I copied this from the olympia patch that added soft deletion - I don't know if its strictly required.